spec/features/admin/merchants/index_spec.rb

```diff
@@ -99,5 +102,62 @@
             expect(current_path).to eq(admin_merchant_path(@brown_merchant.id))
          end
       end
+
+      # User Story 27
+      it 'can enable or disable a merchant from the admin merchants index' do
+         visit admin_merchants_path
+
+         within "#merchant-#{@green_merchant.id}" do
+            expect(page).to have_button("Change Status") 
+            expect(page).to have_content("Enabled")
+         end
+         expect(current_path).to eq(admin_merchants_path)
+
+         within "#merchant-#{@black_merchant.id}" do
+            expect(page).to have_button("Change Status") 
+            expect(page).to have_content("Disabled")
+         end
+         expect(current_path).to eq(admin_merchants_path)
+
+         within "#merchant-#{@brown_merchant.id}" do
+            expect(page).to have_button("Change Status") 
+            expect(page).to have_content("Enabled")
+         end
+         expect(current_path).to eq(admin_merchants_path)
+      end
+      
+      # User Story 28 
+      it 'groups merchants by status' do 
+         visit admin_merchants_path
+
+         expect(page).to have_content("Enabled Merchants")
+         expect(page).to have_content("Disabled Merchants")
+
+         within "#enabled_merchants" do 
+            expect(page).to have_content(@green_merchant.name)
+            expect(page).to have_content(@brown_merchant.name)
+         end
+
+         within "#disabled_merchants" do 
+            expect(page).to have_content(@black_merchant.name)
+         end
+      end
+
+      # User Story 29 
+      it 'shows a link to create new merchant' do
```

Comment:
A good idea is to test the sad path here as well
`fill in("merchant[name]", with: " ")`
`expect(current_path) to eq(new_admin_path)`
`expect(page).to have_content("Error:")`

Reply:
Thank you 
I will add it!